src/mito/src/engine.rs

```diff
@@ -332,51 +332,53 @@ impl<S: StorageEngine> MitoEngineInner<S> {
         )?;
 
         let table_id = request.id;
-        // TODO(dennis): supports multi regions;
-        assert_eq!(1, request.region_numbers.len());
-        let region_number = request.region_numbers[0];
-        let region_id = region_id(table_id, region_number);
-
-        let region_name = region_name(table_id, region_number);
-        let region_descriptor = RegionDescriptorBuilder::default()
-            .id(region_id)
-            .name(&region_name)
-            .row_key(row_key)
-            .default_cf(default_cf)
-            .build()
-            .context(BuildRegionDescriptorSnafu {
-                table_name,
-                region_name,
-            })?;
+        let table_dir = table_dir(schema_name, table_id);
+        let mut regions = HashMap::with_capacity(request.region_numbers.len());
 
-        let _lock = self.table_mutex.lock().await;
-        // Checks again, read lock should be enough since we are guarded by the mutex.
-        if let Some(table) = self.get_table(&table_ref) {
-            if request.create_if_not_exists {
-                return Ok(table);
-            } else {
-                return TableExistsSnafu { table_name }.fail();
+        for region_number in &request.region_numbers {
+            let region_id = region_id(table_id, *region_number);
+
+            let region_name = region_name(table_id, *region_number);
+            let region_descriptor = RegionDescriptorBuilder::default()
+                .id(region_id)
+                .name(&region_name)
+                .row_key(row_key.clone())
+                .default_cf(default_cf.clone())
+                .build()
+                .context(BuildRegionDescriptorSnafu {
+                    table_name,
+                    region_name,
+                })?;
+
+            let _lock = self.table_mutex.lock().await;
```

Comment:
It might be worth moving the table_mutex lock outside the loop, and creating multiple regions of a table should ideally be an atomic operation.